python-template/{{cookiecutter.placeholder_repo_name}}/tox.ini

```diff
@@ -56,7 +56,7 @@ commands =
     rm -f docs/modules.rst
     make -e -C docs clean
     make -e -C docs html
-    python setup.py check --restructuredtext --strict
+    python setup.py check --restructuredtext
```

Comment:
@feanil: I think you added the `--strict` checking in a variety of places. Can you read the PR description and see if this is ok, or if you found an alternative?

Reply:
Note: Although I noted in the PR description notes that there is a bug in `rstcheck`, I think it might actually work here. The difficulty I've been having is solving the following error in `rstcheck`:

```
An `AttributeError` error occured. This is most propably due to a code block directive (code/code-block/sourcecode) without a specified language. This may result in a false negative for source: 'docs/readme.rst'. See https://rstcheck-core.readthedocs.io/en/latest/faq/#code-blocks-without-language-sphinx for more information.
```

Work in progress is at https://github.com/edx/commerce-coordinator/pull/33.

Reply:
@robrap: Yes, absolutely! @feanil, here you go: https://github.com/edx/commerce-coordinator/pull/34

Reply:
@pshiu: That PR doesn't include your explicit addition of the type of code-block to the existing code blocks. Does that fix the problem?

Reply:
@robrap: I tried to cherry-pick it into the PR above via https://github.com/edx/commerce-coordinator/pull/34/commits/e5acc0779630b31fef3ef3e7f357747fc41d7e35. Unfortunately, no dice:

<details>
<summary>Console output</summary>

```
(cc) commerce-coordinator % git log -1 --oneline
e5acc07 (HEAD -> rdrsc, origin/pshiu/RevertDocsRstStrictCheck) fix: AttributeError due to missing language from code-block directive
(cc) commerce-coordinator % python setup.py check --restructuredtext --strict
running check
warning: check: Duplicate implicit target name: "added".

error: Please correct your package.
```
</details>

Reply:
@pshiu: I don't see the AttributeError issue in https://github.com/edx/commerce-coordinator/runs/7009003549?check_suite_focus=true. Just the duplicate implicit target name. Am I looking at the wrong failure log?

Reply:
@robrap (and cc @feanil): The AttributeError issue is unique to `rstcheck`. `rstcheck` is not used by `python setup.py check --restructuredtext --strict`. In summary, I believe:
* If I use **`rstcheck`**:
  * PR: https://github.com/edx/commerce-coordinator/pull/33
  * I run into an AttributeError issue I do not yet know how to solve.
  * Per its [[readme](https://github.com/rstcheck/rstcheck)], rstcheck "checks syntax of reStructuredText and code blocks nested within it.".
    * Thus, I don't think `rstcheck` provides full coverage for what `python setup.py check --restructuredtext --strict` does.
* If I use **`python setup.py check --restructuredtext --strict`**:
  * PR: https://github.com/edx/commerce-coordinator/pull/34
  * I get a "duplicate implicit target name" warning I need direction to solve. (See PR description for workarounds.)
    * I suspect this is a bug in distutils that treats info messages as build blocking warnings under `--strict`, but hesitated to report it as distutils is going to be deprecated.
    * I think the bug is caused by distutils hacking up a `SilentReporter` that overrides docutil's `Reporter.system_message` but ignores its `report_level`. [compare: [distutils source](https://github.com/python/cpython/blob/main/Lib/distutils/command/check.py#L23-L27) and [docutils source](https://sourceforge.net/p/docutils/code/8900/tree/trunk/docutils/docutils/utils/__init__.py#l192)]
  * Per its `--help`, `--restructuredtext` "checks if long string meta-data syntax are reStructuredText-compliant" [[StackOverflow](https://stackoverflow.com/questions/30328259/what-does-python-setup-py-check-actually-do)].
  * Its unofficially recommended substitute [[cite](https://github.com/pypa/readme_renderer/pull/122)], `twine`, "checks whether your distribution’s long description will render correctly on PyPI" [[docs](https://twine.readthedocs.io/en/stable/index.html#twine-check)].

I did a brief survey of openedx repos and observe we tend to fix this by either:
1. Never having duplicate target names (breaks format of, e.g. CHANGELOG.rst)
2. Live with a failing docs check
3. Omit the entire docs check altogether

Definitely appreciate any guidance on how I can help move this forward.

Reply:
@robrap, @feanil: Ooops, looks like I did solve the AttributeError in https://github.com/edx/commerce-coordinator/pull/33/commits/b90a3634012438c0f7fdddee401e820cc8e4a753 on https://github.com/edx/commerce-coordinator/pull/33. Apologies for the confusion.